src/displayBackend/wayland/WaylandZCopy.cpp

```diff
@@ -317,4 +317,111 @@ void WaylandKms::release()
 	}
 }
 
+/*******************************************************************************
+ * Linux dmabuf
+ ******************************************************************************/
+
+WaylandLinuxDmabuf::WaylandLinuxDmabuf(wl_registry* registry,
+									   uint32_t id, uint32_t version) :
+	WaylandZCopy(registry, id, version),
+	mWlLinuxDmabuf(nullptr)
+{
+	try
+	{
+		init(version);
+	}
+	catch(const std::exception& e)
+	{
+		release();
+
+		throw;
+	}
+}
+
+WaylandLinuxDmabuf::~WaylandLinuxDmabuf()
+{
+	release();
+}
+
+/*******************************************************************************
+ * Public
+ ******************************************************************************/
+
+FrameBufferPtr
+WaylandLinuxDmabuf::createLinuxDmabufBuffer(DisplayBufferPtr displayBuffer,
+											uint32_t width,uint32_t height,
+											uint32_t pixelFormat)
+{
+	lock_guard<mutex> lock(mMutex);
+
+	if (!isPixelFormatSupported(pixelFormat))
+	{
+		throw Exception("Unsupported pixel format", EINVAL);
+	}
+
+	return nullptr;
+}
+
+/*******************************************************************************
+ * Private
+ ******************************************************************************/
+
+void WaylandLinuxDmabuf::sOnModifiers(void *data,
+									  zwp_linux_dmabuf_v1 *zwp_linux_dmabuf,
+									  uint32_t format, uint32_t modifier_hi,
+									  uint32_t modifier_lo)
+{
+	/*
+	 * Modifiers are described at
+	 * https://www.khronos.org/registry/EGL/extensions/EXT/EGL_EXT_image_dma_buf_import_modifiers.txt
+	 *
+	 * We do not support passing modifiers from the frontend to the backend,
+	 * so ignore all formats with modifiers set.
+	 */
+	if (modifier_hi == 0 && modifier_lo == 0)
+		static_cast<WaylandLinuxDmabuf*>(data)->onFormat(format);
```

Comment:
data is nullptr , the behavior is unpredictable. Is it expected behavior ?

Reply:
data is guaranteed by wayland to be non-nullptr, otherwise it is a serious bug